platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/ConfigUtils.java

```diff
@@ -33,62 +38,78 @@ public final class ConfigUtils {
     private ConfigUtils() {}
 
     /**
-     * This method will add all config data records (see {@link ConfigData}) that are on the classpath to the given
-     * builder
+     * Scan all classes in a classpath and register all configuration data types with a configuration builder.
+     *
+     * @param configurationBuilder a configuration builder
+     * @return the configuration builder that was passed as a param (for fluent api)
+     */
+    @NonNull
+    public static ConfigurationBuilder scanAndRegisterAllConfigTypes(
+            @NonNull final ConfigurationBuilder configurationBuilder) {
+        return scanAndRegisterAllConfigTypes(configurationBuilder, Collections.emptySet(), Collections.emptyList());
+    }
+
+    /**
+     * Scan all classes in a classpath and register all configuration data types with a configuration builder.
      *
-     * @param configurationBuilder
-     * 		the builder
-     * @return the builder (for fluent api usage)
+     * @param configurationBuilder a configuration builder
+     * @param packagePrefixes      the package prefixes to scan
+     * @return the configuration builder that was passed as a param (for fluent api)
      */
-    public static ConfigurationBuilder addAllConfigDataOnClasspath(final ConfigurationBuilder configurationBuilder) {
-        try (ScanResult result = new ClassGraph().enableAnnotationInfo().scan()) {
-            final Set<? extends Class<? extends Record>> configDataRecordTypes =
-                    result.getClassesWithAnnotation(ConfigData.class.getName()).stream()
-                            .map(classInfo -> (Class<? extends Record>) classInfo.loadClass())
-                            .collect(Collectors.toSet());
-            configDataRecordTypes.forEach(type -> configurationBuilder.withConfigDataType(type));
-            return configurationBuilder;
+    @NonNull
+    public static ConfigurationBuilder scanAndRegisterAllConfigTypes(
+            @NonNull final ConfigurationBuilder configurationBuilder, @Nullable final String... packagePrefixes) {
```

Comment:
Why do you allow `null` for `packagePrefixes`? If I don't want to specify them, I would call the overloaded method above with one parameter anyway.